src/main/kotlin/com/dprint/services/editorservice/v5/EditorServiceV5.kt

```diff
@@ -0,0 +1,216 @@
+package com.dprint.services.editorservice.v5
+
+import com.dprint.core.Bundle
+import com.dprint.services.NotificationService
+import com.dprint.services.editorservice.EditorProcess
+import com.dprint.services.editorservice.EditorService
+import com.dprint.services.editorservice.FormatResult
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.project.Project
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.apache.lucene.util.ThreadInterruptedException
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.atomic.AtomicInteger
+import kotlin.concurrent.thread
+
+private val LOGGER = logger<EditorServiceV5>()
+private var messageId = AtomicInteger(0)
+private const val SLEEP_TIME = 500L
+private const val FORCE_DESTROY_DELAY = 1000L
+
+@Service
+class EditorServiceV5(project: Project) : EditorService {
+
+    private var editorProcess = EditorProcess(project)
+    private var stdoutListener: Thread? = null
+    private val pendingMessages = PendingMessages()
+    private val notificationService = project.service<NotificationService>()
+
+    @Suppress("TooGenericExceptionCaught")
+    private fun createStdoutListener(): Thread {
+        val runnable = Runnable {
+            while (true) {
+                if (Thread.interrupted()) {
+                    return@Runnable
+                }
+                try {
+                    val messageId = editorProcess.readInt()
+                    val messageType = editorProcess.readInt()
+                    val bodyLength = editorProcess.readInt()
+                    val body = MessageBody(editorProcess.readBuffer(bodyLength))
```

Comment:
I think the bug is here. This function internally calls `writeInt`, but this reader thread should now exclusively read from stdout instead of also writing to stdin. I believe here you can now more simply just essentially do `editorProcess.getProcess().inputStream.readNBytes(bodyLength)`.

Reply:
Ah awesome, I will give that a try today